src/libsyncengine/propagation/executor/executorworker.cpp

```diff
@@ -816,76 +763,154 @@ bool ExecutorWorker::checkLiteSyncInfoForCreate(SyncOpPtr syncOp, const SyncPath
     return true;
 }
 
-bool ExecutorWorker::createPlaceholder(const SyncPath &relativeLocalPath) {
+ExitCode ExecutorWorker::createPlaceholder(const SyncPath &relativeLocalPath, ExitCause &exitCause) {
+    exitCause = ExitCause::Unknown;
+
     SyncFileItem syncItem;
     if (!_syncPal->getSyncFileItem(relativeLocalPath, syncItem)) {
         LOGW_SYNCPAL_WARN(_logger, L"Failed to retrieve SyncFileItem associated to item: "
                                            << Utility::formatSyncPath(relativeLocalPath).c_str());
-        _executorExitCode = ExitCode::DataError;
-        _executorExitCause = ExitCause::Unknown;
-        return false;
+        exitCause = ExitCause::InvalidSnapshot;
+        return ExitCode::DataError;
+    }
+    if (!_syncPal->vfsCreatePlaceholder(relativeLocalPath, syncItem)) {
```

Comment:
There seems to be a large overlap with the code which starts at line 866 below. Can this be factored out?

Reply:
Done